hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -657,7 +658,10 @@ int MDMParser::waitFinalResp(_CALLBACKPTR cb /* = NULL*/,
                             if ((r >= 4) && (c != (int)0xFFFFFFFF))  _net.cgi.cell_id  = c;
                             // access technology
                             if (r >= 5) {
-                                _net.act = toCellularAccessTechnology(d);
+                                if(modemIsSaraRxFamily() && d == -1 ) {
```

Comment:
This is probably ok to include R410, just in case?  At a minimum though we should have an R510 check here.

NIT: spacing for `if (`

Reply:
I thought it was cleaner to just use `modemIsSaraRxFamily()` but I can change it to `(_dev.dev == DEV_SARA_R510)` if you would like

Reply:
Yeah since this is a R510 bug, we should probably not alter R410 behavior.